src/v2/api/dtos/QueryAllUsersDTO.ts

```diff
@@ -0,0 +1,28 @@
+import { IsEnum, IsOptional } from 'class-validator';
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { State } from '@prisma/client';
+import { QueryAllDTO } from '../../utils/QueryAllDTO';
+import { validationOptionsMsg } from '../../utils/GLOBALS';
+import { SortQARParam } from './SortQARParam';
+import { SortQAUParam } from './SortQAUParam';
+import { Transform } from 'class-transformer';
+
+export class QueryAllUsersDTO extends QueryAllDTO {
+    @ApiPropertyOptional({
+      enum: SortQARParam,
+      description: 'Sorting by field',
+      default: 'displayName',
+    })
+    @IsEnum(SortQAUParam, validationOptionsMsg('Sort must be an enum'))
+    @IsOptional()
+      sort?: SortQAUParam;
+
+    @ApiPropertyOptional({
+      description: 'State of the user',
+      enum: State,
+    })
+    @Transform(({ value }) => value.split(','))
+    @IsOptional()
+      state?: State[];
```

Comment:
now you don't validate parameter, you should add @Validate or add validation in Pipe